### ISSUE TITLE
eas completely ignores hooks field from app.json

### ISSUE THREAD

**Issue by RohovDmytro** [expo/eas-cli#897]
### Build/Submit details page URL

https://expo.dev/accounts/rohovdmytro/projects/houserion/builds/a833de11-ecee-4877-a607-91b994468125

### Summary

````
{
  "expo": {
    "name": "Housy",
    "backgroundColor": "red",
    "icon": "./assets/app-icon-houserion.png",
    "slug": "houserion",
    "privacy": "hidden",
    "entryPoint": "index.js",
    "platforms": ["android", "ios"],
    "hooks": {
      "postPublish": [
        {
          "file": "sentry-expo/upload-sourcemaps",
          "config": {
            "organization": "universage",

            "project": "houserion",
            "authToken": "..."
          }
        }
      ]
    },
  },
...

}
````

### Managed or bare?

bare

### Environment

 EAS CLI 0.43.0 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.13.0 - /usr/local/bin/node
      Yarn: 1.22.15 - /usr/local/bin/yarn
      npm: 8.1.0 - /usr/local/bin/npm
    Utilities:
      Git: 2.30.1 - /usr/bin/git
    npmPackages:
      expo: ^44.0.4 => 44.0.4 
      expo-updates: ^0.11.3 => 0.11.3 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3
    Project workflow: generic

### Error output

There is no error, but I do not see my source maps and release being created in Sentry.

### Reproducible demo or steps to reproduce from a blank project

Sorry, the project is too complicated.

**Comment by RohovDmytro** [comment]
This is what EAS interface shows me, no hooks field:

![image](https://user-images.githubusercontent.com/1548235/148587949-de01f1a8-64b4-4961-b34e-1a9433cf864d.png)

**Comment by dsokal** [comment]
The `postPublish` hook is not executed when building a project. EAS Build has nothing to do with updates.

**Comment by jvgaeta** [comment]
You're ignoring the android.config and ios.config fields as well, which is required for using maps.

https://docs.expo.dev/versions/latest/sdk/map-view/

```
Copy your API Key into your app.json under the android.config.googleMaps.apiKey field.
Rebuild the app binary and re-submit to Google Play or sideload it (depending on how you configured your API key) to test that the configuration was successful.
```

**Comment by Darmolar** [comment]
> You're ignoring the android.config and ios.config fields as well, which are required for using maps.
> 
> https://docs.expo.dev/versions/latest/sdk/map-view/
> 
> ```
> Copy your API Key into your app.json under the android.config.googleMaps.apiKey field.
> Rebuild the app binary and re-submit to Google Play or sideload it (depending on how you configured your API key) to test that the configuration was successful.
> ```

I am having the same issue 
please anyone with the solution

**Comment by juspeeh** [comment]
If anyone comes across this thread. There has been an issue with eas build for android which is solved by updating sentry-expo to 5.0.3. Here are more details: https://forums.expo.dev/t/eas-build-does-not-execute-postpublish-hook/66790
